### PR TITLE
Enable demo-loop app test in CI

### PR DESCRIPTION
**Pull request by paulromano**
This PR enables the demo-loop app test to run in CI and fixes a few issues @vrpascuzzi and I have encountered with recent builds:
- HepMC3 spits out some diagnostic output that interferes with the expected JSON output in simple-driver.py.
- The storage factor for secondaries is not sufficiently high to complete all steps in the demo-loop app. Bumping it up to 10 seems to work on my setup :crossed_fingers:

### CODE REVIEW

File: app/CMakeLists.txt [hunk 1]
```diff
@@ -246,8 +245,12 @@ if(CELERITAS_BUILD_DEMOS AND CELERITAS_USE_VecGeom)
       ENVIRONMENT "${_env};${_geant_test_env}"
       RESOURCE_LOCK gpu
       REQUIRED_FILES "${_driver};${_gdml_inp};${_hepmc3_inp}"
-      DISABLED true
     )
+    if(NOT CELERITAS_USE_Geant4 OR NOT CELERITAS_USE_ROOT OR NOT CELERITAS_USE_HepMC3)
+      set_tests_properties("app/demo-loop" PROPERTIES
+        DISABLED false
```
Comment (sethrj) [inline app/CMakeLists.txt:251]: Shouldn't this be `DISABLED true`? By default the test is enabled, but if any of these dependencies is missing then we disable the test.

File: src/io/EventReader.cc [hunk 2]
```diff
@@ -22,8 +24,14 @@ EventReader::EventReader(const char* filename, SPConstParticles params)
...
 {
     CELER_EXPECT(params_);
+
+    // Turn off/capture HepMC3 diagnostic output that pollutes our own output
+    HepMC3::Setup::set_debug_level(-1);
+    ScopedStreamRedirect silenced(&std::cout);
+
     // Determine the input file format and construct the appropriate reader
     input_file_ = HepMC3::deduce_reader(filename);
+    CELER_LOG(diagnostic) << "HepMC3 said: " << silenced.str();
```
Comment (sethrj) [inline src/io/EventReader.cc:34]: If `set_debug_level` works as expected, `silenced` will be empty and this log message will be a bit silly.  Maybe
```suggestion
    std::string output = silenced.str();
    if (!output.empty())
    {
        CELER_LOG(diagnostic) << "HepMC3 said: " << output;
    }
```

**Comment by sethrj** [comment]
Oh noooooo, HepMC3 is using `printf` instead of `cout`, so I don't think that `ScopedStreamRedirect` will work 😭

**Comment by sethrj** [comment]
Also we're getting an amazing `cuda error: unspecified launch failure` when building debug.

File: app/CMakeLists.txt (hunk 1 above)
Comment (paulromano) [reply app/CMakeLists.txt:251]: :facepalm: yup

**Comment by sethrj** [comment]
@paulromano To get this working with the goofy hepmc3 printf statements, how about we just filter the output in python to ignore everything before the first output line that starts with `{`?

**Comment by sethrj** [comment]
And to get the unit tests working, let's also disable the test if `CMAKE_BUILD_TYPE STREQUAL "Debug"`...

**Comment by paulromano** [comment]
Sounds easy enough -- I'll add the filtering to simple-driver.py. Do you want me to get rid of the `ScopedStreamRedirect`?

**Comment by sethrj** [comment]
Yeah, we might as well... sorry that didn't work! Thanks for the fixes.

**Comment by paulromano** [comment]
@sethrj Can you restart the CI job? It looks like it failed due to docker pull rate limits but should be fine now.

**Comment by sethrj** [comment]
@paulromano I don't think there's a straightforward way for me to do that... can you just do `git commit --allow-empty -m "Trigger CI test"` and push?

**Comment by paulromano** [comment]
Strangely that doesn't seem to have fixed it. We're still seeing a message about docker pull rate limits even though it's been well more than six hours. Any ideas?

**Comment by paulromano** [comment]
@sethrj Tests are passing now after retriggering.

File: app/demo-loop/simple-driver.py [hunk 3]
```diff
@@ -57,6 +57,8 @@
 
 print("Received {} bytes of data".format(len(result.stdout)))
 out_text = result.stdout.decode()
+# Filter out spurious HepMC3 output
+out_text = out_text[out_text.find('\n{') + 1:]
```
Comment (sethrj) [inline app/demo-loop/simple-driver.py:61]: This might cause newer HepMC3 versions (which don't write to stdout) to fail...

Comment (paulromano) [reply app/demo-loop/simple-driver.py:61]: I did think about that case too -- in the case where the string `\n{` is not in the output (as happens when HepMC3 doesn't spit anything out), `out.text.find(...)` returns -1, so it ends up being `out_text[0:]`.

Comment (sethrj) [reply app/demo-loop/simple-driver.py:61]: Sneaky!

**Review by sethrj — APPROVED** [review]

**Pull request merged**